hangman.rb

```diff
@@ -0,0 +1,125 @@
+
+require 'colorize'
+
+class Hangman
+
+ 
+
+  def initialize
+    @user_guesses = []
+    @wrong_count = 0
+    @word = pick_word 
+    puts "Answer: #{@word.join}"
+    @underlines = create_underlines
+    @ascii =  ["|     _________", 
+               "|     |/      |", 
+               "|     |",
+               "|     |",
+               "|     |",
+               "|     |",
+               "|     |",
+              ]
+
+    @man_array =  [
+                  "|     |      / \\".colorize(:red),
+                  "|     |       |".colorize(:cyan),
+                  "|     |      \\|/".colorize(:light_yellow),
+                  "|     |      (_)".colorize(:magenta)
+                  ]
+
+    play
+  end
+
+#Program picks a word for the player.
+  def pick_word
+    word_options = %w(scared selection shake shaking shallow shout silly simplest slight
+                    slip slope soap solar species spin stiff swung tales thumb zoo
+                    tobacco toy trap treated tune university vapor vessels wealth wolf)
+
+    word = word_options.sample.split(//)
+    return word
+  end
+#joins the ascii art and the underlines for the letters in the word.
+  def ascii_art
+    @ascii[7] = @underlines.join
+    puts @ascii
+  end
+#creates the correct amount of underlines for each letter in the word.
+  def create_underlines
+    underlines = ["| ____|___ word: "]
+    @word.length.times do
+      underlines.push("_ ")
+    end
+    return underlines
+  end
+#adds pieces to the ascii art man each time the user gets answers incorrectly. 
+#Does not penalize user for guessing the same letter  
+  def add_to_man
+    if !@user_guesses.include?(@user_letter.upcase)
+      @wrong_count += 1
+      index_to_replace_ascii = @wrong_count + 1
+      new_ascii_row = @man_array.pop
+      @ascii[index_to_replace_ascii] = new_ascii_row
+    else
+      puts "You already guessed this letter"
+    end
+  end
+#replaces the underlines with correct letters.
+  def replace_underlines
+    @word.each_with_index do |letter, index|
+      if @user_letter == letter
+        @underlines[index+1] = @user_letter.upcase + " "
```

Comment:
Even when referring to an array index, you want to use whitespace around the `+` `[index + 1]`